### PR TITLE
[19.05] Interactive Environment Fixes

### PR DESCRIPTION
**Pull request by dannon**
With the removal of the (long deprecated) static libs (in this case, specifically requirejs, toastr, and jquery) staged for external consumption during this past cycle IE setup was broken.

This PR adjusts all IES to use bundled libraries via the generic Galaxy entrypoint we provide, be requirejs-free, and along the way takes several refactoring steps towards providing a single pluggable interface that hopefully (will target dev, not a release branch with this) will eventually allow creation of new IES without writing custom javascript (and hopefully mako, eventually) at all.

~WIP for a few while I follow up on Toastr, but this is enough to get IEs functional as-is.~

### CODE REVIEW

**Review by mvdbeek — APPROVED** [review]
Great, works for me! The rstudio issue should be resolved with https://github.com/erasche/docker-rstudio-notebook/pull/23

**Comment by hexylena** [comment]
Merged and rebuilt, @mvdbeek look ok now?

**Comment by dannon** [comment]
@erasche Thanks, I'll re-fetch and test it now.  Going to push one more minor tweak in a sec to the docs here to mention host.docker.internal magic alias as the galaxy_infra_url.

(It worked great)

File: config/galaxy.yml.sample [hunk 1]
```diff
@@ -693,11 +693,10 @@ galaxy:
   # URL (with schema http/https) of the Galaxy instance as accessible
   # within your local network - if specified used as a default by pulsar
   # file staging and Jupyter Docker container for communicating back
-  # with Galaxy via the API.  If you are attempting to setup GIEs on Mac
-  # OS X with Docker for Mac - this should likely be the IP address of
-  # your machine on the virtualbox network (vboxnet0) setup for the
-  # Docker host VM. This can found by running ifconfig and using the IP
-  # address of the network vboxnet0.
+  # with Galaxy via the API.  If you are attempting to set up GIEs on
+  # Mac OS X with Docker Desktop for Mac this should be
```
Comment (mvdbeek) [inline config/galaxy.yml.sample]: I'd maybe change this to:
```
and your Galaxy instance runs on port 8080 this should be
```
I guess it is obvious to us, but maybe that'll help some folks ?

Comment (dannon) [reply config/galaxy.yml.sample]: @mvdbeek Sure thing, good idea.

**Comment by mvdbeek** [comment]
Thanks @dannon, looks very good. Test failure was unrelated, we may just need to give Jenkins a little more time.

**Pull request merged**